pkg/sql/create_view.go

```diff
@@ -92,25 +91,29 @@ func (n *createViewNode) startExec(params runParams) error {
 	// First check the backrefs and see if any of them are temporary.
 	// If so, promote this view to temporary.
 	backRefMutables := make(map[descpb.ID]*tabledesc.Mutable, len(n.planDeps))
+	hasTempBackref := false
 	for id, updated := range n.planDeps {
 		backRefMutable := params.p.Descriptors().GetUncommittedTableByID(id)
 		if backRefMutable == nil {
 			backRefMutable = tabledesc.NewExistingMutable(*updated.desc.TableDesc())
 		}
-		if !persistence.IsTemporary() && backRefMutable.Temporary {
-			// This notice is sent from pg, let's imitate.
-			params.p.BufferClientNotice(
-				params.ctx,
-				pgnotice.Newf(`view "%s" will be a temporary view`, viewName),
-			)
-			persistence = tree.PersistenceTemporary
+		if !n.persistence.IsTemporary() && backRefMutable.Temporary {
+			hasTempBackref = true
 		}
 		backRefMutables[id] = backRefMutable
 	}
+	if hasTempBackref {
+		n.persistence = tree.PersistenceTemporary
```

Comment:
Note: this was missing and was a bug, we need to update the persistence. Is it bad that I mutate `n.persistence` instead of `persistence`?

Reply:
Also, I added a test in `tabledesc` to cover this, search for `// Temporary tables.`.

Reply:
The `temp_table` logic test fails here: https://github.com/postamar/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/temp_table#L143
with
```
CREATE VIEW upgrade_temp_view AS SELECT a FROM temp_table;
	-- FAIL
E210224 16:02:27.451156 3907 sql/logictest/logic.go:3563  [-] 2  
E210224 16:02:27.451156 3907 sql/logictest/logic.go:3563  [-] 2 +testdata/logic_test/temp_table:143: CREATE VIEW upgrade_temp_view AS SELECT a FROM temp_table
E210224 16:02:27.451156 3907 sql/logictest/logic.go:3563  [-] 2 +expected success, but found
E210224 16:02:27.451156 3907 sql/logictest/logic.go:3563  [-] 2 +(XXUUU) relation "upgrade_temp_view" (71): referenced schema descriptor 65 not found
E210224 16:02:27.451156 3907 sql/logictest/logic.go:3563  [-] 2 +validate.go:221: in GetSchemaDescriptor()
```
This happens because the view is on a temp table and therefore should also have its persistence set to `tree.PersistenceTemporary` or else during validation it will try to cross-reference a non-existent schema. Previously the persistence was _not_ set because further down in this file we create the table descriptor using `n.persistence` instead of the local `persistence` variable.